### PR TITLE
Farm prices bugfix

### PR DESCRIPTION
**Pull request by fedestraus**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

### CODE REVIEW

**Comment by fedestraus** [comment]
Hello fellow Apes,

In this commit the LP pricing uses the multicallfetching with a similar function than the token prices (I have to create a new one though because I changed the previous a little bit).

For the config file (lpPrices.ts) I assumed that:

1) all LP Tokens have 18 decimals
2) farm pid with no name are deprecated or not used, therefore I did not copy them.

Let me know your thoughts.

Regards

Fede

File: src/hooks/api.ts [hunk 1]
```diff
@@ -332,3 +332,10 @@ export const getHeadersHome = async () => {
 
   return headers
 }
+
+export const getFarmPrices = async () => {
+  const url = "https://api.apeswap.finance/stats/farmPrices";
+  const resp = await fetch(url);
+  const data = await resp.json();
+  return data;
```
Comment (ape-swap) [inline src/hooks/api.ts:340]: Thanks a lot for your contribution!

Should we delete this unused piece of code?

Comment (fedestraus) [reply src/hooks/api.ts]: It won't be used indeed. I already deleted it for you.

My apologies!

Cheers

Fede

**Review by Harambe-Nakamoto — CHANGES_REQUESTED** [review]
Nice work! Would it be possible for you to extend the original tokenPrices code to handle LP tokens with a separate call? There is a lot of code you could reuse with that.

**Comment by fedestraus** [comment]
> Nice work! Would it be possible for you to extend the original tokenPrices code to handle LP tokens with a separate call? There is a lot of code you could reuse with that.

I didn't want to change the original tokenPrices function. But I completely agree with your comment. Will make the changes and commit them ASAP

**Comment by fedestraus** [comment]
I've refactored the fetch prices function to handle both tokens and Lp tokens fetch pricing.

Let me know your thoughts on that and tell me if this fulfills your request.

Fede

File: src/hooks/api.ts (hunk 1 above)
Comment (ape-swap) [reply src/hooks/api.ts]: Thanks!

**Review by Harambe-Nakamoto — CHANGES_REQUESTED** [review]
Could you leave the lpPrice call out of the farms state and use the same technique as tokenPrices? Pretty much what you had before, but reusing the tokenPrice calls. You can grab the LP address from the farm config instead of creating another file for them. Let me know if that makes sense.

File: src/state/lpPrices/fetchLpPrices.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    const multicallContract = getContract(multicallABI, multicallContractAddress, chainId)
+    const apePriceGetterAddress = getApePriceGetterAddress(chainId)
+    const tokensToCall = Object.keys(farmsConfig).filter((token) => farmsConfig[token].lpAddresses[chainId] !== undefined)
+    const erc20Calls = tokensToCall.map((token) => {
+        return {
+            address: farmsConfig[token].lpAddresses[chainId],
+            name: 'decimals',
+        }
+    })
+    const tokenDecimals = await multicall(multicallContract, erc20ABI, erc20Calls)
```
Comment (Harambe-Nakamoto) [inline src/state/lpPrices/fetchLpPrices.ts:21]: We actually dont need to fetch token decimals for LPs because they will be 18 decimals

**Review by Harambe-Nakamoto — CHANGES_REQUESTED** [review]
Nice work! Just one last small change and its good to go.

**Review by Harambe-Nakamoto — APPROVED** [review]

**Pull request merged**